File: backend/src/main/java/com/example/SEPDrive/controller/rideRequestDTO.java
package com.example.SEPDrive.controller;
import com.example.SEPDrive.model.carClass;

import java.util.List;


public record rideRequestDTO(double distance, double duration, double price, LatLng start, String startaddress, List<LatLng> zwischenstops, List<String> zwischenstopssaddress, LatLng destination, String destinationaddress, carClass carClass){

    }

File: frontend/Angular/src/app/models/rideRequestDTO.ts
export interface LatLng {
  lat: number;
  lng: number;
}

export interface ZwischenStopp {
  name: string;
  latlng: LatLng;
  ispassed: boolean;
}

export interface edit{
  destination?: LatLng;
  destinationaddress?: string;
  zwischenstops?: LatLng[];
  zwischenstopssaddress?: string[];
  cost?: number;
  duration?: number;
  distance?: number;
}

export interface rideRequestDTO {
  distance: number;
  duration: number;
  price: number;
  start: LatLng;
  startaddress: string;
  zwischenstops: LatLng[];
  zwischenstopssaddress: string[];
  destination: LatLng;
  destinationaddress: string;
  carClass: 'klein' | 'Medium' | 'Deluxe';

}
